loki/transformations/pool_allocator.py

```diff
@@ -624,8 +623,9 @@ def _create_stack_allocation(self, stack_ptr, stack_end, ptr_var, arr, stack_siz
             dim = Product((dim, _dim))
         arr_type_bytes = InlineCall(Variable(name='C_SIZEOF'),
                                             parameters=as_tuple(self._get_c_sizeof_arg(arr)))
-        arr_type_bytes = InlineCall(function=Variable(name='MAX'),
-                    parameters=(arr_type_bytes, Literal(8)), kw_parameters=())
+        if not any(s in dim for s in self.horizontal.size_expressions):
```

Comment:
Yes, good spot, this could really do with a comment. Thanks!

The minimal padding could certainly be made optional, but for IFS components we wouldn't add any safety by padding everything and we would hurt performance a lot in single precision builds. So even if it is made optional, I think it should be enabled by default. Whilst I agree that not relying on the `horizontal` dimension does make the transformation more general, I don't think already quite complex transformation needs another option 😄 

I'll leave the final call on whether to make it optional up to you.

Reply:
As discussed offline and even though I am aware that adapting the tests will be a bit of a torture ... I think you/we should make this another option :grimacing: Thank you ...